adoc/chapters/cuda_backend.adoc

```diff
@@ -0,0 +1,598 @@
+
+Some core SYCL features require a minimum compute capability for the CUDA
+backend.
+
+[[table.extensionsupport]]
+.CUDA support for Core SYCL API features
+[width="100%",options="header",cols="33%,33%,33%"]
+|====
+| Feature                                   | SYCL Aspect               | Required Compute Capability 
+| [code]#16-bit floating point#             | [code]#aspect::fp16#      | 5.3 or greater
+|====
+
+[[sec:non_core_features_and_extensions]]
+== Non-core features and extensions
+
+Some additional functions are provided for the CUDA backend in the
+`sycl::cuda` namespace.
+
+[[table.noncorefeatures]]
+.CUDA support for non-Core SYCL APIs
+[width="100%",options="header",cols="33%,33%,33%"]
+|====
+| API                                                    | Description
+| [code]#bool sycl::cuda::has_native_event(sycl::event)# | Returns `true` if the SYCL event has a valid `CUevent` that can be queries via application interop.
+|====
+
+Additional CUDA features are available depending upon the device's compute 
+capability. SYCL can support these optional CUDA features with extensions.
+
+Use of CUDA extensions requires that the API for a given extension is available
+to the SYCL implementation. This needs to be determined at compile time. 
+Checking for the existence of feature test macros is the preferred method
+for checking whether an API exists. The feature test macro format 
+is `SYCL_EXT_<vendor>_<feature>`. The `<vendor>` string may also contain the
+word `CUDA` for features specific to CUDA. For example, the feature test macro
+for CUDA extensions in oneAPI may be either `SYCL_EXT_ONEAPI_CUDA_<feature>`,
+or just `SYCL_EXT_ONEAPI_<feature>`.
+
+Use of a given CUDA extension also requires that a chosen device has the
+required compute capability to use the CUDA extension. This can be determined
+using `sycl::aspect`s. Non-core SYCL aspects may be defined by an
+implementation which would allow this check to happen at runtime.
+
+The table below shows a proposal for SYCL supported CUDA extensions. This should
+be populated by other members of the SYCL community.
+
+[[table.extensionsupport]]
+.SYCL support for CUDA 11.3 extensions
+[width="100%",options="header",cols="35%,35%,15%, 15"]
+|====
+| CUDA Extension                            | SYCL Aspect   | Feature Test Macro               | Required Compute Capability 
+|====
+
+[[sub:cuda:builtin-kernel-functions]]
+=== Built-in Kernel Functions
+The CUDA backend specification currently does not define any built-in kernel 
+functions.
+
+[[sub:cuda:error_handling]]
+=== Error Handling
+
+SYCL uses `sycl::errc` as an enum class to hold error codes. These error
+codes may originate in the SYCL runtime or be passed from other runtimes to
+the SYCL runtime. When a `sycl::exception` is thrown, the `sycl::errc` can
+be queried using the exception's `.code()` method. Possible values for 
+`sycl::errc` include: `success`, `runtime`, `memory_allocation`, and more.
```

Comment:
I wonder if we want this last sentence.  It seems to place an unnecessary dependency between this interop spec and the core SYCL spec.  Is it important to list the specific `sycl::errc` error codes?

Reply:
I also think it is not important to list the specific `sycl::errc` codes in this way and the sentence can be removed.

Reply:
This has been dealt with in our internal draft, commit here will be made once other points are dealt with.